### PR TITLE
TIP-1143: Create Product Model Query

### PR DESCRIPTION
**Pull request by pierallard**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | y
| Added legacy Behats               | n
| Added acceptance tests            | n
| Added integration tests           | y
| Changelog updated                 | n
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php [hunk 1]
```diff
@@ -71,9 +70,6 @@ class ProductController
     /** @var PaginatorInterface */
     protected $searchAfterPaginator;
 
-    /** @var ParameterValidatorInterface */
-    protected $parameterValidator;
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php]: Link to your other PR those removal?

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/ApplyProductSearchQueryParametersToPQB.php [hunk 2]
```diff
@@ -56,14 +62,13 @@ public function apply(
 
         foreach ($searchParameters as $propertyCode => $filters) {
             foreach ($filters as $filter) {
-                $searchLocale = $query->searchLocaleCode;
-                $context['locale'] = isset($filter['locale']) ? $filter['locale'] : $searchLocale;
+                $context['locale'] = isset($filter['locale']) ? $filter['locale'] : $searchLocaleCode;
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/ApplyProductSearchQueryParametersToPQB.php]: I do prefer the use of `??` much more readable

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/ApplyProductSearchQueryParametersToPQB.php [hunk 3]
```diff
@@ -56,14 +62,13 @@ public function apply(
...
 
                 if (isset($filter['locales']) && '' !== $filter['locales']) {
                     $context['locales'] = $filter['locales'];
                 }
 
-                $context['scope'] = isset($filter['scope']) ? $filter['scope'] : $query->searchChannelCode;
+                $context['scope'] = isset($filter['scope']) ? $filter['scope'] : $searchChannelCode;
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/ApplyProductSearchQueryParametersToPQB.php]: Same here

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/Validator/ListProductModelsQueryValidator.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        $this->validateGrantedAttributes = $validateGrantedAttributes;
+        $this->validateGrantedLocales = $validateGrantedLocales;
+    }
+
+    /**
+     * @param ListProductModelsQuery $query
+     *
+     * @throws InvalidQueryException
+     */
+    public function validate(ListProductModelsQuery $query): void
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/Validator/ListProductModelsQueryValidator.php]: missing spec

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/ListProductModelsQueryHandler.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+use Akeneo\Tool\Component\Api\Exception\InvalidQueryException;
+use Akeneo\Tool\Component\Api\Pagination\PaginationTypes;
+use Akeneo\Tool\Component\Api\Security\PrimaryKeyEncrypter;
+use Akeneo\Tool\Component\StorageUtils\Cursor\CursorInterface;
+use Akeneo\Tool\Component\StorageUtils\Exception\PropertyException;
+
+/**
+ * @author Pierre Allard <pierre.allard@akeneo.com>
+ */
+class ListProductModelsQueryHandler
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/UseCase/ListProductModelsQueryHandler.php]: missing spec (and also for product :disappointed: )

**Pull request merged**